### ISSUE TITLE
x/build/cmd/watchflakes: failures posted with misleading timestamps

### ISSUE THREAD

**Issue by bcmills** [golang/go#62116]
In https://github.com/golang/go/issues/62095#issuecomment-1682426774, `watchflakes` posted a number of failures in `x/` repos with timestamps dating back to June, like:
> 2023-06-21 11:18 linux-amd64-racecompile mod@baa5c2d0 go@2fcfdb96 go/internal/gccgoimporter [build] ([log](https://build.golang.org/log/d21cecb0c6bbcca25a9c44445379ab875970da57))

That timestamp is misleading: `go@2fcfdb96` was committed `2023-08-17`, not `2023-06-21`.
(The timestamp is for `mod@baa5c2d0`, ignoring the `go` commit time.)

Those kinds of misleading timestamps can easily cause an issue to be mistriaged: a sudden new failure starting two months ago is probably a misconfigured or recently-added builder, whereas a sudden widespread recent failure is almost certainly a bad commit that should be rolled back.

(attn @cherrymui)

**Comment by bcmills** [comment]
https://github.com/golang/go/issues/63730#issuecomment-1779486991 is another one with misleading timestamps, which would prevent it from being correctly matched to #63739.

**Comment by cherrymui** [comment]
Given the current data model with the coordinator, this is rather difficult to do. LUCI will provide us better data. As we're going to switch to LUCI, we will solve it with LUCI.

**Comment by bcmills** [comment]
I agree that the coordinator makes this difficult. FWIW, `fetchlogs` uses `maintner` for this information:
https://cs.opensource.google/go/x/build/+/master:cmd/fetchlogs/fetchlogs.go;l=143-158;drc=a3f1e410a493bb04cbc55bb7c99b81457f6cae66

It's not the most efficient (`maintner` is kind of a memory hog and takes a short while to load from scratch), but it at least sidesteps the need to make changes to the coordinator.

If `watchflakes` has enough network access to use `maintner`, I expect this could have a workaround in place with less than a day's work.
